### ISSUE TITLE
[Bankster Testing] SigningService

### ISSUE THREAD

**Issue by rwrzesien** [golemfactory/concent#742]
This is part of #647. It depends on #740.

- Create one wrapper which:
    - Creates a socket and connects it to given component.
    - Creates a given `Frame` with given payload.
    - Uses `MiddleMan` protocol for sending `Frame` over created socket.
    - Uses `MiddleMan` protocol for receiving response.
    - Verifies that response is as expected.

If https://github.com/golemfactory/concent/issues/743 was implemented already, reuse the wrapper.

Create a test suite for testing communication flow: request to `SigningService` => response from `SigningService`.

- test that sending `GolemMessageFrame` with correct data in `TransactionSigningRequest` should return `GolemMessageFrame` with `SignedTransaction`.
- test that sending `GolemMessageFrame` with incorrect data in `TransactionSigningRequest` should return `GolemMessageFrame` with `TransactionRejected`.
- test that sending `GolemMessageFrame` with different Golem message than `TransactionSigningRequest` should return `ErrorFrame` with error code `UnexpectedMessage`.

**Comment by dybi** [comment]
@rwrzesien, I am wondering if we need such tests for `SigningService` alone. What I mean is - what would be advantage of such integration tests over already existing "unit-integration tests"?

**Comment by rwrzesien** [comment]
@dybi The experience has shown that we had a lot of unit-integration tests for `ConcentAPI`, but when we run them from 'outside' not everything was working as expected ;) The advantage might be that this tests can be run against different environments.

Also, a lot of work here is already done. Most of these tests will have the same logic and prepare data in the same way as current unit-integration tests. What will probably change is the removal of all mocks, and usage of some wrapper to send and receive data. I think that when wrappers will be done, doing this issue shouldn't take much.

We can make this lowest priority.

**Comment by PaweuB** [comment]
Pivotal:
https://www.pivotaltracker.com/story/show/159927384

**Comment by rwrzesien** [comment]
Closing, as this will be technically not possible to test `SigningService` alone, without mimic most of the `MiddleMan` functionality, which will be duplicate of https://github.com/golemfactory/concent/issues/743.

**Comment by PaweuB** [comment]
A few questions:
1) Maybe it should be tested along with MiddleMan? 
2) How it should be tested if it's closed issue?
